src/overlays/effects/ovl_Effect_Ss_En_Ice/z_eff_ss_en_ice.c

```diff
@@ -1,30 +1,157 @@
+void EffectSsEnIce_UpdateFlying(PlayState* play, u32 index, EffectSs* this);
+void EffectSsEnIce_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsEnIce_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
 const EffectSsInit Effect_Ss_En_Ice_InitVars = {
     EFFECT_SS_EN_ICE,
     EffectSsEnIce_Init,
 };
 
-#endif
+u32 EffectSsEnIce_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsEnIceInitParams* initParams = PARAMS;
+
+    if (initParams->type == ENICE_TYPE_FLYING) {
+        Math_Vec3f_Copy(&this->pos, &initParams->pos);
+
+        if (initParams->actor != NULL) {
+            Math_Vec3f_Diff(&this->pos, &initParams->actor->world.pos, &this->vec);
+        } else {
+            Math_Vec3f_Copy(&this->vec, &initParams->pos);
+        }
+
+        Math_Vec3f_Copy(&this->velocity, &gZeroVec3f);
+        Math_Vec3f_Copy(&this->accel, &gZeroVec3f);
+        this->life = 10;
+        this->actor = initParams->actor;
+        this->draw = EffectSsEnIce_Draw;
+        this->update = EffectSsEnIce_UpdateFlying;
+        this->rScale = initParams->scale * 100.0f;
+        this->rPrimColorR = initParams->primColor.r;
+        this->rPrimColorG = initParams->primColor.g;
+        this->rPrimColorB = initParams->primColor.b;
+        this->rPrimColorA = initParams->primColor.a;
+        this->rEnvColorR = initParams->envColor.r;
+        this->rEnvColorG = initParams->envColor.g;
+        this->rEnvColorB = initParams->envColor.b;
+        this->rAlphaMode = 1;
+        this->rPitch = randPlusMinusPoint5Scaled(0x10000);
+    } else if (initParams->type == ENICE_TYPE_NORMAL) {
+        Math_Vec3f_Copy(&this->pos, &initParams->pos);
+        Math_Vec3f_Copy(&this->vec, &initParams->pos);
+        Math_Vec3f_Copy(&this->velocity, &initParams->velocity);
+        Math_Vec3f_Copy(&this->accel, &initParams->accel);
+        this->life = initParams->life;
+        this->draw = EffectSsEnIce_Draw;
+        this->update = EffectSsEnIce_Update;
+        this->rLifespan = initParams->life;
+        this->rScale = initParams->scale * 100.0f;
+        this->rYaw = Math_FAtan2F(initParams->velocity.z, initParams->velocity.x);
+        this->rPitch = 0;
+        this->rPrimColorR = initParams->primColor.r;
+        this->rPrimColorG = initParams->primColor.g;
+        this->rPrimColorB = initParams->primColor.b;
+        this->rPrimColorA = initParams->primColor.a;
+        this->rEnvColorR = initParams->envColor.r;
+        this->rEnvColorG = initParams->envColor.g;
+        this->rEnvColorB = initParams->envColor.b;
+        this->rAlphaMode = 0;
+    } else {
+        return 0;
+    }
+
+    return 1;
+}
+
+void EffectSsEnIce_Draw(PlayState* play, u32 index, EffectSs* this) {
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+    s32 pad;
+    f32 scale = this->rScale * 0.01f;
+    u32 gameplayFrames = play->gameplayFrames;
+    f32 alpha;
+
+    OPEN_DISPS(gfxCtx);
+
+    if (this->rAlphaMode != 0) {
```

Comment:
Should `rAlphaMode` be boolean?

Reply:
Not quite sure, it probably makes more sense as an enum (though only two members so look boolean like), but it looks like they did boolean operations on it. I think I would like to come back to this when I document it more fully.